XenModel/Actions/Network/CreateSriovAction.cs

```diff
@@ -78,22 +77,45 @@ protected override void Run()
                 selectedPifs.Insert(0, pifOnMaster);
             }
 
-            int inc = 100 / (selectedPifs.Count + 1);
-            int lo = inc;
+            int inc = 100 / selectedPifs.Count;

```

Comment:
As I mentioned in my previous comment, we also need to ensure Count is not 0.
Perhaps the best is to check and if 0 return - after all, if we have no pifs, do we need to create the network?

Reply:
I think in NewNetworkWizard.cs we already check the pif is not 0 ```if (sriovSelectedPifs.Count != 0)
                (new CreateSriovAction(xenConnection, network, sriovSelectedPifs)).RunAsync();```
Do we need check here again?

Reply:
I recommend we check again; it makes the code much safer if something changes in the way the action is called.

Reply:
It's a cheap check anyway without any practical impact on performance.

Reply:
OK, I agree it's safer. Thank you.